### DISCUSSION TITLE
How on news page component to get resulting data of other component calling?

### DISCUSSION THREAD

**Discussion by sergeynilov** [livewire/livewire#8548] [1. Help]
In laravel 10 / "livewire/livewire 3.4 app in app/Livewire/Cacheable/HotNews.php component I made class for calculating and 
grouping data and keeping them in cache of the app:

    <?php
    
    namespace App\Livewire\Cacheable;
    
    use App\Enums\ConfigValueEnum;
    use App\Enums\NewsPublishedEnum;
    use App\Library\Services\Interfaces\KeyDataInterface;
    use App\Library\Traits\{AppCommonTrait, DataCachingTrait};
    use App\Models\KeyData;
    use App\Models\News;
    use App\Models\Reaction;
    use Illuminate\Support\Facades\DB;
    use Illuminate\Support\Str;
    use Livewire\Component;
    
    class HotNews extends Component
    {
        public $reactions = [];
        use AppCommonTrait, DataCachingTrait;
    
        protected string $keyDataIdentifier;
        protected int $companyId;
        public int $cachingTimeInSeconds = 60 * 10;
        public bool $writeDataCaching = false;
        public bool $useDataCaching = false;
        public string $cachedData = '';
    
        public function mount(int $companyId)
        {
            $this->companyId = $companyId;
            $this->keyDataIdentifier = 'hot-news-cache-by-company-' . $this->companyId;
            $this->cachingTimeInSeconds = 60 * (int)ConfigValueEnum::get(ConfigValueEnum::CACHED_HOT_NEWS_BLOCK_MINUTES);
            $this->useDataCaching = ConfigValueEnum::get(ConfigValueEnum::USE_DATA_CACHING, false); // Reference to USE_DATA_CACHING parameter in .env file
            $this->cachedData = $this->readDataCaching($this->keyDataIdentifier);
            if (!$this->useDataCaching or empty($this->cachedData)) { // if cache clear - need read data from db and save in cache
                $this->retrieveDbData();
                $this->writeDataCaching = $this->useDataCaching;
            }
        }
    
        public function render(): string
        {
            if($this->cachedData) { // There is cached data
                return $this->convertCachingData($this->cachedData);
            }
    
            // If has no cached data - needs to generate it from blade view
            $renderedCode = view('livewire.Cacheable.hot-news', ['reactions' => $this->reactions])->render();
            if($this->writeDataCaching) { // Use cached data
                $this->keyDataService->write(keyDataIdentifier: $this->keyDataIdentifier,
                    cachingTimeInSeconds: $this->cachingTimeInSeconds,
                    data: $renderedCode);
            }
            return $renderedCode;
        }
    
        protected function retrieveDbData(): void
        {
            $newsTb = (new News)->getTable();
            $reactionTb = (new Reaction)->getTable();
    
            // Get more news reactions count grouped by news
            $listingLimit = ConfigValueEnum::get(ConfigValueEnum::HOT_NEWS_LISTING_LIMIT);
            $this->reactions = Reaction
                ::getByModelType(MODEL_TYPE_NEWS)
                ->leftJoin($newsTb, $newsTb . '.id', '=', $reactionTb . '.model_id')
                ->whereHas('model', function ($query) use ($newsTb) {
                    $query->where($newsTb . '.company_id', $this->companyId);
                    $query->where($newsTb . '.published', NewsPublishedEnum::PUBLISHED);
                })
                ->groupBy('model_id')
                ->orderBy('reaction_count', 'desc')
                ->limit($listingLimit)
                ->select(
                    $reactionTb . '.model_id',
                    DB::raw('count(' . $reactionTb . '.id) as reaction_count'))
                ->get()
                ->reject(function (Reaction $reaction) {
                    $news = News::find($reaction->model_id);
                    return empty($news);
                })
                ->map(function (Reaction $reaction) {
                    $news = News::find($reaction->model_id);
                    if ( ! empty($news)) {
                        $reaction->news_id = $news->id;
                        $reaction->news_slug = $news->slug;
                        $reaction->news_title = $news->title;
                        $reaction->published_at = $news->created_at;
                    }
    
                    return $reaction;
                });
        }
    }

It works ok with refresh period 10 minutes and I call it in blade file as :

    <livewire:Cacheable.hot-news :companyId="$selectedCompany->id" lazy />

also this component is used inside of other news page component, which has refresh period  in 30 days.

This news page component has code with calling of HotNews component

    
    <div class="frontend_subtitle d1">
        <livewire:Cacheable.hot-news :companyId="$newsItem->company_id" lazy />
    </div>
    

The calling of HotNews is wrapped with a CHEABLE_HOT_NEWS_BLOCK tags and I want to cut it with fresh data from component HotNews calling?
But for this I need manually in news page component to get string value of Cacheable.hot-news.
In other words how on news page component to get resulting data of other component HotNews calling ?
